File: Irrgarten_Ruby/test_p1.rb
#encoding:utf-8

require_relative 'weapon'
require_relative 'shield'
require_relative 'dice'
require_relative 'game_state'
require_relative 'irrgarten'

module Irrgarten 
	class TestP1
	
		def self.prueba_enum
			puts "Probando dirección: #{Directions::LEFT}"
			puts "Probando dirección: #{Directions::RIGHT}"
			puts "Probando dirección: #{Directions::UP}"
			puts "Probando dirección: #{Directions::DOWN}"
			
			puts "Probando orientación: #{Orientation::VERTICAL}"
			puts "Probando orientación: #{Orientation::HORIZONTAL}"
			
			puts "Probando personajes: #{GameCharacter::PLAYER}"
			puts "Probando personajes: #{GameCharacter::MONSTER}"
		end
		
		def self.prueba_weapon
			w = Weapon.new(2.0, 3)
			puts "\n"
			puts w.to_s	
			4.times do
				puts "Potencia de disparo #{w.attack}"
				puts w.to_s
				puts "¿Se descarta el arma? #{w.discard}"
			end
		end
		
		def self.prueba_shield
			s = Shield.new(2.0,3)
			puts "\n"
			puts s.to_s
			4.times do
				puts "Daño protegido #{s.protect}"
				puts s.to_s
				puts "¿Se descarta el escudo? #{s.discard}"
			end
		end
		
		def self.prueba_dice
			for i in 1..100
				puts "\nPrueba #{i}"
				puts "Posición: #{Dice.random_pos(10)}"
				puts "Empieza jugador: #{Dice.who_starts(3)}"
				puts "Inteligencia: #{Dice.random_intelligence}"
				puts "Fuerza: #{Dice.random_strength}"
				puts "¿Resucita? #{Dice.resurrect_player}"
				puts "Armas conseguidas: #{Dice.weapons_reward}"
				puts "Escudos conseguidos: #{Dice.shields_reward}"
				puts "Salud conseguida: #{Dice.health_reward}"
				puts "Poder arma: #{Dice.weapon_power}"
				puts "Poder escudo: #{Dice.shield_power}"
				usos = Dice.uses_left
				puts "Usos restantes: #{usos}"
				puts "Intensidad: #{Dice.intensity(3.0)}"
				puts "¿Se descarta? #{Dice.discard_element(usos)}"
			end
		end
		
		def self.prueba_game_state
			g = GameState.new("laberintoooo", "jugadoreees", "monstruitoss", 1, false, "logeate")
			puts "\n"
			puts g.labyrinth
			puts g.players
			puts g.monsters
			puts g.current_player
			puts g.winner
			puts g.log
		end
	end
end

Irrgarten::TestP1.prueba_enum
Irrgarten::TestP1.prueba_weapon
Irrgarten::TestP1.prueba_shield
Irrgarten::TestP1.prueba_dice
Irrgarten::TestP1.prueba_game_state

File: Irrgarten_Java/Irrgarten_P4/src/irrgarten/TestP1.java

package irrgarten;

/**
 *
 * @author leonin04
 */
public class TestP1 {
    static void pruebaEnum (){
        for (Directions dir : Directions.values()) {
            System.out.println("Probando dirección: " + dir);
        }
        
        for (Orientation ori : Orientation.values()) {
            System.out.println("Probando orientación: " + ori);
        }
        
        for (GameCharacter charac : GameCharacter.values()) {
            System.out.println("Probando personajes: " + charac);
        }

    }

    static void pruebaWeapon (float daño, int usos){
        Weapon arma1 = new Weapon(daño,usos);
        System.out.println("Dañó: " + arma1.attack());
        
        System.out.println(arma1.toString());
        
        System.out.println("¿Se descarta? " + arma1.discard());
    }
    
    static void pruebaShield(float durabilidad, int usos) {
        Shield escudo1 = new Shield(durabilidad,usos);
        System.out.println("Protegio: " + escudo1.protect());
        
        System.out.println(escudo1.toString());
        
        System.out.println("¿Se descarta? " + escudo1.discard());

    }
    
    static void pruebaDice(){
        
        for (int i=0; i<100; i++){
            System.out.println("\nPrueba " + (i+1));
            System.out.println("Posición: " + Dice.randomPos(10));
            System.out.println("Empieza jugador " + Dice.whoStarts(3));
            System.out.println("Inteligencia " + Dice.randomIntelligence());
            System.out.println("Fuerza " + Dice.randomStrength());
            System.out.println("¿Resucita? " + Dice.resurrectPlayer());
            System.out.println("Armas conseguidas: " + Dice.weaponsReward());
            System.out.println("Escudos conseguidos: " + Dice.shieldsReward());
            System.out.println("Salud conseguida: " + Dice.healthReward());
            System.out.println("Poder arma: " + Dice.weaponPower());
            System.out.println("Poder escudo: " + Dice.shieldPower());
            int usos = Dice.usesLeft();
            System.out.println("Usos restantes: " + usos);
            System.out.println("Intensidad: " + Dice.intensity(3.0f));
            System.out.println("¿Se descarta? " + Dice.discardElement(usos));
        }
    }
    
    static void pruebaGameState(){
        GameState game = new GameState("laberintooo", "jugadoresee", "monstruooos", 1, false, "logeee");
        System.out.println(game.getLabyrinth());
        System.out.println(game.getPlayers());
        System.out.println(game.getMonsters());
        System.out.println(game.getCurrentPlayer());
        System.out.println(game.getWinner());
        System.out.println(game.getLog());
    }
    
     public static void main(String[] args) {
       System.out.println("Probando enum:");
       pruebaEnum();
       
       System.out.println("\n Probando arma:");
       pruebaWeapon(10.0f,1);
       pruebaWeapon(10.0f,0);
       
       System.out.println("\n Probando escudo:");
       pruebaShield(10.0f,1);
       pruebaShield(10.0f,0);
       
       System.out.println("\n Probando dado:");
       pruebaDice();
       
       System.out.println("\nProbando GameState:");
       pruebaGameState();

    }
}
